ChopChop/Models/Ingredients/Quantity.swift

```diff
@@ -0,0 +1,258 @@
+/**
+ Represents the quantity of an ingredient.
+ 
+ Invariants:
+ - Quantities are non negative.
+ 
+ Mass type quantities are meaasured in kilograms.
+ Volume type quantities are measured in litres.
+ */
+enum Quantity {
+    case count(Double)
+    case mass(Double)
+    case volume(Double)
+
+    var type: QuantityType {
+        switch self {
+        case .count:
+            return .count
+        case .mass:
+            return .mass
+        case .volume:
+            return .volume
+        }
+    }
+
+    var value: Double {
+        get {
+            switch self {
+            case .count(let value):
+                return value
+            case .mass(let value):
+                return value
+            case .volume(let value):
+                return value
+            }
+        }
+        set {
+            switch self {
+            case .count:
+                self = .count(newValue)
+            case .mass:
+                self = .mass(newValue)
+            case .volume:
+                self = .volume(newValue)
+            }
+        }
+    }
```

Comment:
Given that you have to do this, might it not be a better idea to have a `Quantity` class instead, with a `quantityType` and `value`? My quantity enum wasn't actually meant to be used for the actual quantities, I was intending to map from whatever `Quantity` objecf you had.

Reply:
I think this idea might solve a lot of the duplication problems, I'll try redesigning it

Reply:
I'm going ahead with the redesign for now but a small problem I just thought of is how to deal with a new `Quantity` type whose `value` is not a `Double`

Reply:
I imagine since the value has to be a number, a `Double` probably covers all possible cases.